frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx

```diff
@@ -58,6 +73,34 @@ const UserProfileSidebar: React.FC<Properties> = ({
     void dispatch(authActions.signOut());
   }, [dispatch]);
 
+  const handleClose = useCallback((): void => {
+    dialogReference.current?.open && dialogReference.current.close();
+  }, [dialogReference]);
+
+  const handleOpen = useCallback(() => {
+    setCurrentStepIndex(INITIAL_STEP);
+    dialogReference.current?.showModal();
+  }, [dialogReference]);
+
+  const goToNextStep = useCallback((): void => {
+    setCurrentStepIndex((previous) => {
+      return Math.min(
+        previous + NEXT_STEP_INCREMENT,
+        STEPS.length - NEXT_STEP_INCREMENT,
+      );
+    });
+  }, []);
+
+  const getCurrentStep = (): ReactElement | null => {
+    if (!currentStep) {
+      return null;
+    }
```

Comment:
```suggestion
```

Reply:
for this one I need to check that the current exists otherwise, the typescript throws an error:

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/107894952/8d817eec-0d1f-4d6f-a1ed-65f42b4332d5)

I updated the function like this to avoid the if statement but I am not sure it complies with what is requested, if it is not can you give me a suggestion to avoid this typescript error? 

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/107894952/2cf7c2a5-5638-4b7a-86f7-6f8a0c98d439)